fs/operations/operations.go

```diff
@@ -361,10 +361,21 @@ func Copy(ctx context.Context, f fs.Fs, dst fs.Object, remote string, src fs.Obj
 	if !ci.Inplace && f.Features().Move != nil && f.Features().PartialUploads && !strings.HasSuffix(remote, ".rclonelink") {
 		// Avoid making the leaf name longer if it's already lengthy to avoid
 		// trouble with file name length limits.
-		suffix := "." + random.String(8) + ".partial"
+		ext := strings.TrimSpace(ci.PartialUploadExtension)
+		if len(ext) == 0 {
```

Comment:
I don't think you need this `if len(ext) == 0 { ... }` clause do you as the context.Background() Config wasn't what the user asked for here.

Reply:
Yeah. If it will be `suffix` and not `extension` than it definitely can be zero-length
Previous logic with file ends with `*.RNDSTRNG.` just had no sense